### PR TITLE
Add keep_newlines argument to wrap_text

### PR DESCRIPTION
**Pull request by theFox6**
If keep_newlines is true, every new line given in the input text is also contained in the output text.

### CODE REVIEW

**Comment by theFox6** [comment]
the previous function arguments can still be used in the same way

**Comment by paramat** [comment]
I'm wondering why?

**Comment by theFox6** [comment]
?

File: builtin/common/misc_helpers.lua [hunk 1]
```diff
@@ -312,8 +312,7 @@ function core.formspec_escape(text)
 	return text
 end
 
-
-function core.wrap_text(text, max_length, as_table)
+function core.linebreaker function(text, max_length)
```
Comment (sfan5) [inline builtin/common/misc_helpers.lua]: :question:

**Comment by rubenwardy** [comment]
> I'm wondering why?

It's closer to how wrapping works in most things.

File: builtin/common/misc_helpers.lua (hunk 1 above)
Comment (SmallJoker) [reply builtin/common/misc_helpers.lua]: Code tested :100: / :100: 

---
If you want to keep this function public (`minetest`, `core` table), then please document it.

**Comment by paramat** [comment]
Oh i see now, sorry.

**Comment by theFox6** [comment]
> If you want to keep this function public (minetest, core table), then please document it.

Where and how should I document it?

**Comment by ClobberXD** [comment]
@theFox6: `core.line_break()` should be documented in `lua_api.txt` under the [_Helper functions_](https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L2501) section as `minetest.line_break()`.

**Comment by theFox6** [comment]
got it now

**Comment by theFox6** [comment]
It's because I didn't want to change the behaviour of `mintest.wrap_text()`.

**Comment by ClobberXD** [comment]
Oh right... good point.

Will test in a day or two.

File: builtin/common/misc_helpers.lua [hunk 2]
```diff
@@ -312,12 +312,11 @@ function core.formspec_escape(text)
 	return text
 end
 
-
-function core.wrap_text(text, max_length, as_table)
+function core.line_break function(text, max_length)
```
Comment (sfan5) [inline builtin/common/misc_helpers.lua]: you still haven't fixed the obvious syntax error here
you should test that your PR works correctly

**Comment by theFox6** [comment]
yeah, sorry I should really do this locally instead of using the web editor

**Comment by SmallJoker** [comment]
And you should better use `git rebase` locally instead of using the web function, because it seems to create a merge commit (??). Strictly spoken this PR does not need rebase, but squashing your changes would be nice to have.

**Comment by sfan5** [comment]
Have you tested your code?

**Comment by theFox6** [comment]
I've tested it, it works.

**Comment by ClobberXD** [comment]
It'd be very helpful for testers if test mods are provided with PRs that modify the Lua API. I don't even know how / where to use `wrap_text`... :confused:

**Comment by theFox6** [comment]
@ClobberXD My [Journal](https://forum.minetest.net/viewtopic.php?f=11&t=19611) modpack uses `wrap_text`.

**Comment by ClobberXD** [comment]
@theFox6 Some irrelevant commits seem to have been added to this PR...

**Comment by theFox6** [comment]
Somebody wanted me to rebase it...

**Comment by theFox6** [comment]
I don't know what I'm doing wrong but a rebase seems to create a merging commit

**Pull request closed without merging**